bin/node/src/chain_spec.rs

```diff
@@ -104,17 +95,15 @@ pub struct AuthorityKeys {
 
 #[derive(Debug, StructOpt, Clone)]
 pub struct ChainParams {
-    /// Pass the chain id.
-    ///
-    /// It can be a predefined one (dev) or an arbitrary chain id passed to the genesis block
-    /// `dev` chain id means that a set of known accounts will be used to form a comittee
-    #[structopt(long, value_name = "CHAIN_SPEC", default_value = "a0dnet1")]
+    /// Chain ID is a short identifier of the chain
+    #[structopt(long, value_name = "ID", default_value = "a0dnet1")]
     pub chain_id: String,
 
-    #[structopt(long, default_value_if("chain-id", Some(DEVNET_ID), "true"), conflicts_with_all(&["account-ids", "sudo-account-id"]))]
-    pub dev_chain: bool,
+    /// The type of the chain. Can be "dev", "local", "live" or custom (any other string)
+    #[structopt(long, value_name = "TYPE", default_value = CHAINTYPE_LIVE)]
```

Comment:
smells like an Enum here. Or just always run a live chain, as close to the real thing as possible?

Reply:
I would go for an `Enum`. Then you can probably use this macro: https://docs.rs/clap/2.29.0/clap/macro.arg_enum.html to avoid implementing `FromStr` by hand.

And I really do not see the need for a custom variant. Is there a reason for that?

Reply:
About `enum` - I don't see any reason to define a custom `enum` when `ChainType` is already here. I could change the type of the argument from `String` to `ChainType` and provide a custom parser, but that solution doesn't really differ from what we have now (https://github.com/h4nsu/aleph-node/commit/824966b90779ec64a8e98c855983a17dd64656e2)

About `clap` macro - I'd rather avoid that solution because a) it requires defining another enum b) it mixes `clap` and `structopt` in a way I don't fully understand

Reply:
fair point with `clap`;

I am a little bit against the "custom" type, because any typo / error with chain type leads to a custom type (like the one with the typo from another comment).

Reply:
I don't think there's really any functional difference between `Live` and `Custom` (or `Development` and `Local`), this whole chaintype seems to be more of a label for external use (modulo different network flags for `{Develompent,Local}` vs. `{Live,Custom}`)

Reply:
I got rid of the custom variant, now it panics for values other than live, dev or local